File: app/src/main/java/com/mutant/godutch/BaseActivity.java
package com.mutant.godutch;

import android.content.pm.ActivityInfo;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;
import com.mutant.godutch.model.Friend;

/**
 * Created by jackie780919 on 2016/2/16.
 */
public abstract class BaseActivity extends AppCompatActivity {

    public abstract int getLayoutId();

    public abstract void findViews();

    public abstract void setup();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(getLayoutId());
        overridePendingTransition(0, 0);
        findViews();
        setup();
        setOrientationLand();
    }

    protected void setOrientationLand() {
        if (getRequestedOrientation() == ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE) {
            setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);
        }
    }

    @Override
    public void finish() {
        super.finish();
        overridePendingTransition(0, 0);
    }

    @Nullable
    protected Friend getMeInFriend() {
        FirebaseUser firebaseUser = FirebaseAuth.getInstance().getCurrentUser();
        if(firebaseUser != null) {
            String mePhotoUrl = "";
            if(firebaseUser.getPhotoUrl() != null) {
                mePhotoUrl = firebaseUser.getPhotoUrl().toString();
            }

            return new Friend(firebaseUser.getUid(), "Me", mePhotoUrl);
        }
        return null;
    }
}

File: app/src/main/java/com/mutant/godutch/BaseActivity.kt
package com.mutant.godutch

import android.content.pm.ActivityInfo
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.Toolbar
import android.view.View
import com.google.firebase.auth.FirebaseAuth
import com.mutant.godutch.model.Friend

/**
 * Created by jackie780919 on 2016/2/16.
 */
abstract class BaseActivity : AppCompatActivity() {

    abstract val layoutId: Int

    abstract fun setup()

    internal var mToolbar: Toolbar? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(layoutId)
        overridePendingTransition(0, 0)
        setupToolbar()
        setup()
        setOrientationLand()
    }

    internal fun hideToobar() {
        findViewById(R.id.tool_bar)?.visibility = View.GONE
    }

    internal fun setupToolbar() {
        mToolbar = findViewById(R.id.tool_bar) as Toolbar?
        setSupportActionBar(mToolbar)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        supportActionBar?.setDisplayShowHomeEnabled(true)
    }

    internal fun setupToolbar(rootView: View) {
        mToolbar = rootView.findViewById(R.id.tool_bar) as Toolbar?
        setSupportActionBar(mToolbar)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        supportActionBar?.setDisplayShowHomeEnabled(true)
    }

    private fun setOrientationLand() {
        if (requestedOrientation == ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE) {
            requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
        }
    }

    override fun onSupportNavigateUp(): Boolean {
        onBackPressed()
        return true
    }

    override fun finish() {
        super.finish()
        overridePendingTransition(0, 0)
    }

    val me: Friend
        get() {
            val firebaseUser = FirebaseAuth.getInstance().currentUser
            if (firebaseUser != null) {
                var mePhotoUrl = ""
                if (firebaseUser.photoUrl != null) {
                    mePhotoUrl = firebaseUser.photoUrl!!.toString()
                }

                return Friend(firebaseUser.uid, firebaseUser.displayName!!, mePhotoUrl)
            }
            return Friend()
        }
}
